Frameworks/Foundation/NSString.mm

```diff
@@ -537,6 +537,10 @@ - (BOOL)getBytes:(void*)buffer
  @Status Interoperable
 */
 - (void)getCharacters:(unichar*)dest range:(NSRange)range {
+    if (range.location + range.length > self.length) {
```

Comment:
We don't exactly want that here; if length is 5, `NSLocationInRange(5, {1,2})` will be false but that doesn't make this invalid.